provider/ansible.py

```diff
@@ -176,6 +169,15 @@ def distro_install():
         path.find_command('ansible-playbook')
     except path.CmdNotFoundError:
         ansible_install_policy = params['ansible_install_policy']
-        return policy_map[ansible_install_policy]()
-    else:
-        return True
+        if not policy_map[ansible_install_policy]():
+            return False
+
+    # Install ansible depended packages that can't be installed
+    # by pip (or are not a dependency) when installing ansible
+    package_list = params.get_list("package_list", 'sshpass')
+    ansible_install_policy = 'distro_install'
+    for package in package_list:
+        if not policy_map[ansible_install_policy](package):
+            return False
```

Comment:
The packages installation method is not related with the `ansible_install_policy`, this policy is used for ansible only.
Suggest:
```
from virttest import utils_package
...
package_list = params.get_list("package_list", 'sshpass')
if not utils_package.package_install(package_list):
    return False
```

Reply:
I see, but we still need to use `distro_install` method, because it handles addition of custom repositories, which is exactly the case of `sshpass` (at least in case of RHEL 8.5). At least `ansible_install_policy` can be renamed, so it is not confusing in this context.

Reply:
You are right, we need `sshpass` ~
From the existing approach, when use customer repo to `distro_install` the `ansiable`,  package `sshpass` can be installed automatically as a dependency.  This is the reason that it does not specify `sshpass` to be installed after setting `ansible_repo`.

Reply:
Yep, actually the only problem necessary to address is in relation with the pip installed ansible :slightly_smiling_face:

Reply:
Em~  call `distro_install` will setup/remove repo multiple times,  how about:
```
    package_list = params.get_list("package_list", 'sshpass')
    try:
        path.find_command('ansible-playbook')
    except path.CmdNotFoundError:
        ansible_install_policy = params['ansible_install_policy']
        if ansible_install_policy == 'distro_install':
            package_list.append('ansible')
        else:
            if not policy_map[ansible_install_policy]():
                return False


    if not policy_map['distro_install'](package_list):
        return False
    # If ansible and dependents packages are installed correctly
    return True
```

As use package_list, the function `distro_install` need be updated accordingly. omit here.

Reply:
@yanan-fu I like your idea :slightly_smiling_face: This is a good point regarding repository. 

I prepared the change: `utils_package.package_install` accepts package name or list of packages, so the change of `distro_install` was quite minor.